publisher-sdk/src/main/java/com/criteo/publisher/util/jsonadapter/ArrayListJsonAdapter.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.util.jsonadapter
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import java.io.IOException
+import java.lang.reflect.Type
+import java.util.Arrays
+
+internal abstract class ArrayListJsonAdapter<C : MutableCollection<T>?, T> private constructor(
+    private val elementAdapter: JsonAdapter<T>
+) : JsonAdapter<C>() {
+  abstract fun newCollection(): C
+
+  @Throws(IOException::class)
+  override fun fromJson(reader: JsonReader): C {
+    val result = newCollection()
+    reader.beginArray()
+    while (reader.hasNext()) {
+      result?.add(elementAdapter.fromJson(reader)!!)
+    }
+    reader.endArray()
+    return result
+  }
+
+  @Throws(IOException::class)
+  override fun toJson(writer: JsonWriter, value: C?) {
+    writer.beginArray()
+    for (element in value!!) {
+      elementAdapter.toJson(writer, element)
+    }
+    writer.endArray()
+  }
+
+  override fun toString(): String {
+    return "$elementAdapter.collection()"
+  }
+
+  companion object {
+    @JvmStatic
+    val factory = Factory { type, annotations, moshi ->
+      val rawType = Types.getRawType(type)
+      if (annotations.isNotEmpty()) return@Factory null
```

Comment:
Could you explain why do you return null for this condition ?

Reply:
This class is taken from moshi code see the same condition [here](https://github.com/square/moshi/blob/master/moshi/src/main/java/com/squareup/moshi/CollectionJsonAdapter.kt#L51)